service.go

```diff
@@ -286,6 +287,8 @@ func (s *serviceManager) Process(env *network.Envelope) {
 // closeDatabase closes the database.
 // It also removes the database file if the path is not default (i.e. testing config)
 func (s *serviceManager) closeDatabase() error {
+	s.dbMutex.Lock()
```

Comment:
This seems like it is only protecting the write of "s.db = nil" below. How about if we just remove that, and leave db set. Other goroutines that try to act on s.db will then get some kind of "error already closed" error from boltdb: https://godoc.org/github.com/boltdb/bolt#ErrDatabaseNotOpen